apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerReporter.java

```diff
@@ -115,7 +115,7 @@ public Thread newThread(Runnable r) {
     @Override
     public void report(Transaction transaction) {
         if (!tryAddEventToRingBuffer(transaction, TRANSACTION_EVENT_TRANSLATOR)) {
-            transaction.recycle();
+            transaction.decrementReferences();
```

Comment:
Also won't be needed if spans are created with reference count of 0 and ended implicitly when references go down to 0